### PR TITLE
Feature different day horoscope

### LINKED ISSUE #4: Yesterday Functionality
Make a Fetch call for the yesterday button when on the horoscope page to populate with yesterday's information for that sign

### LINKED ISSUE #5: Tomorrow Functionality
Make a Fetch call for the tomorrow button when on the horoscope page to populate with tomorrow's information for that sign

### LINKED ISSUE #32: User can view different day horoscopes
As a user,
When on a horoscope page,
I should be able to toggle between today, tomorrow, and yesterday horoscopes.

Acceptance Criteria:
You should be able to select Yesterday or Tomorrow buttons on the initial horoscope page and see the information change. If not on today's you should see a button for Today's horoscope. Clicking it will populate with the information for today.

### PR DESCRIPTION
**Pull request by lbmerchant93**
What does this PR do?
Display horoscope information on horoscope page
Include functionality for today, tomorrow, and yesterday buttons
Store state on app and pass information through props
DDAA

What does it feature/fix?
Feature the horoscope information that is fetched from the api depending on the sign selected and the date chosen, default is today.

Where should the reviewer start?
App.js line 14

How should this be tested?
Should be able to select a sign, see the correct sign's today horoscope, then can use the buttons to toggle between today, tomorrow, and yesterday.

What are the next steps or any issues/concerns?
Create functionality to look up what sign you are, save your favorite signs
Testing

### CODE REVIEW

File: src/Components/App/App.js [hunk 1]
```diff
@@ -5,22 +5,72 @@ import Header from '../Header/Header';
...
+      horoscope: '',
+      image: ''
+    }
+  }
+
+  retrieveHoroscopeData = async (url, src, alt, when) => {
+    await fetchHoroscope(url)
+    .then(result => {
+        this.setState({ isLoading: true, horoscope: '' })
+        if (typeof result === 'string') {
```
Comment (kristenmb) [inline src/Components/App/App.js:27]: Cool use of settimeout here I think the loading message adds a lot to the vibe and styling of the app!

File: src/Components/App/App.js [hunk 2]
```diff
@@ -5,22 +5,72 @@ import Header from '../Header/Header';
...
+                    horoscope: result,
+                    isLoading: false
+                })
+            }, 600)   
+          }
+    })
+  }
 
+  retrieveDifferentDay = (when) => {
+    let url = `https://aztro.sameerkumar.website/?sign=${this.state.sign}&day=${when}`;
```
Comment (kristenmb) [inline src/Components/App/App.js:49]: Is there a way you could store the various urls in the apiCalls file as variables and dynamically access them through what you pass into the api call? might clean this up a little!

Comment (kristenmb) [inline src/Components/App/App.js:49]: Is there a way you could store the various urls in the apiCalls file as variables and dynamically access them through what you pass into the api call? might clean this up a little!

File: src/Components/App/App.js [hunk 3]
```diff
@@ -5,22 +5,72 @@ import Header from '../Header/Header';
...
+      image: ''
+    }
+  }
+
+  retrieveHoroscopeData = async (url, src, alt, when) => {
+    await fetchHoroscope(url)
+    .then(result => {
+        this.setState({ isLoading: true, horoscope: '' })
+        if (typeof result === 'string') {
+            setTimeout(() => {
```
Comment (kristenmb) [inline src/Components/App/App.js:28]: Cool use of settimeout here I think the loading message adds a lot to the vibe and styling of the app!

File: src/Components/Home/Home.css [hunk 4]
```diff
@@ -4,4 +4,10 @@
     flex-wrap: wrap;
     column-gap: 20px;
     row-gap: 20px;
+    animation: fadeIn ease 1.5s;
 }
+
+@keyframes fadeIn {
```
Comment (kristenmb) [inline src/Components/Home/Home.css:10]: Love the fade in, it is so calming.

File: src/Components/Home/Home.js [hunk 5]
```diff
@@ -14,25 +14,25 @@ import libra from'../../images/libra.png';
...
-        {src: aries, alt: 'Aries'},
-        {src: taurus, alt: 'Taurus'},
-        {src: gemini, alt: 'Gemini'}, 
-        {src: cancer, alt: 'Cancer'}, 
-        {src: leo, alt: 'Leo'},
-        {src: virgo, alt: 'Virgo'},
-        {src: libra, alt: 'Libra'},
-        {src: scorpio, alt: 'Scorpio'}, 
-        {src: sagittarius, alt: 'Sagittarius'}
+        {src: capricorn, alt:'capricorn'},
```
Comment (kristenmb) [inline src/Components/Home/Home.js:19]: If you wanted to move this array into a separate data file you definitely could, but its not huge so I dont know if you need to. Just a potential nice to have future refactor!

File: src/Components/Horoscope/Horoscope.js [hunk 6]
```diff
@@ -1,66 +1,38 @@
...
+    return (
+        <main className="horoscope-main">
+            {errorMsg && <p className='error-message'>{errorMsg}</p>}
+            {isLoading && <p className='loading-message'>Loading...</p>}
+            {horoscope  &&   
+                <section className="horoscope-content">
+                    <img className='horoscope-sign-img' src={image} alt={alt} />
+                    <div className='details'>
+                        <h2 className='horoscope-sign-title'>{sign.toUpperCase()}</h2>
+                            <ul>
```
Comment (kristenmb) [inline src/Components/Horoscope/Horoscope.js:16]: I might break these lines out a bit more to make it more readable. Just by popping the h3 tags onto their own line I think would help a lot!

File: src/Components/Horoscope/Horoscope.js [hunk 7]
```diff
@@ -1,66 +1,38 @@
...
+                                <li className='date-range'><h3>Birthday Range:</h3> {horoscope.date_range}</li>
+                                <li className='days-color'><h3>Color:</h3> {horoscope.color}</li>
+                                <li className='days-compatibility'><h3>Compatibility:</h3> {horoscope.compatibility}</li>
+                               
+                                <li className='days-lucky-numbers'><h3>Lucky Number:</h3> {horoscope.lucky_number}</li>
+                                <li className='days-lucky-time'><h3>Lucky Time:</h3> {horoscope.lucky_time}</li>
+                                <li className='days-mood'><h3>Mood:</h3> {horoscope.mood}</li>
+                                 <li className='days-description'><h3>Description:</h3> {horoscope.description}</li>
+                            </ul>
+                            {day !== 'yesterday' && <button className='yesterday' onClick={() => props.retrieveDifferentDay('yesterday')}>Yesterday</button>}
```
Comment (kristenmb) [inline src/Components/Horoscope/Horoscope.js:27]: This conditional render is super clever!

**Review by kristenmb — COMMENTED** [review]
This is looking really awesome Lucas! Mainly just a couple points of potential refactoring to clean up some files, but I love the way things are tied together and the styling it so polished already!

File: src/Components/App/App.js (hunk 2 above)
Comment (lbmerchant93) [reply src/Components/App/App.js:49]: Little confused on how to do this. This url here is already set to be dynamic and if i set it to a variable, I don't know how to change that variable being dynamic. Not 100% sure on what you're suggesting.

**Pull request merged**